integration-tests/production-runtime/cypress/integration/production.js

```diff
@@ -14,13 +14,23 @@ describe(`Production build tests`, () => {
       .location(`pathname`)
       .should(`equal`, `/page-2/`)
 
-    cy.reload()
-      .go(`back`)
+    cy.reload().go(`back`)
 
     cy.waitForRouteChange()
       .getTestElement(`page2`)
       .should(`exist`)
       .location(`pathname`)
       .should(`equal`, `/`)
   })
+
+  it(`should show 404 page when visiting non-existent page route`, () => {
+    cy.getTestElement(`404`)
+      .first()
+      .click()
+
+    cy.waitForRouteChange()
+      .getTestElement(`404`)
```

Comment:
You could also [check the pathname here](https://github.com/gatsbyjs/gatsby/blob/master/integration-tests/path-prefix/cypress/integration/path-prefix.js#L25), too! But this works just as well!

Reply:
@DSchau thanks for the heads up! We took the `.first()` call out and added assertion for pathname validity.